framework/platform/bpl/include/bpl/bpl_cfg.h

```diff
@@ -256,62 +257,69 @@ int cfg_get_management_mode();
  *
  * @return BPL_CERTIFICATION_MODE_ON Certification mode is ON.
  * @return BPL_CERTIFICATION_MODE_OFF Certification mode is OFF.
+ * @return default_value if value not set
  * @return -1 Error.
  */
-int cfg_get_certification_mode();
+int cfg_get_certification_mode(int default_value = -1);
 
 /**
  * Returns the maximum number of failures allowed on agent before stopping its execution.
  *
  * @return Maximum number of failures allowed or 0 to retry indefinitely.
+ * @return default_value if value not set
  * @return -1 Error.
  */
-int cfg_get_stop_on_failure_attempts();
+int cfg_get_stop_on_failure_attempts(int default_value = -1);
 
 /**
  * Returns whether the platform is in onboarding state.
  *
  * @return 1 Platform is in onboarding state.
  * @return 0 Platform is NOT in onboarding state.
+ * @return default_value if value not set
  * @return -1 Error.
  */
-int cfg_is_onboarding();
+int cfg_is_onboarding(int default_value = -1);
 
 /**
  * Checks the state of the RDKB Extensions feature.
  *
  * @return 1 Enabled.
  * @return 0 Disabled.
+ * @return default_value if value not set
  * @return -1 Error.
  */
-int cfg_get_rdkb_extensions();
+int cfg_get_rdkb_extensions(int default_value = -1);
 
 /**
  * Checks the state of the Band Steering feature.
  *
  * @return 1 Enabled.
  * @return 0 Disabled.
+ * @return default_value if value not set
  * @return -1 Error.
  */
-int cfg_get_band_steering();
+int cfg_get_band_steering(int default_value = -1);
```

Comment:
Default should be false

Reply:
It is 1 in Linux implementation. Should it be put into DB since it is non-default?

Reply:
defaults can be different for different implementations, good point.
In Linux, most should be enabled by default - I would keep everything as it is today.